resources/Hooks/useStore.js

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useState, useEffect } from 'react'
```

Comment:
can you explain how this works when you have a chance?

Reply:
Yeah so the basic flow is to pass an instance of the store via react context so that we can access it by pulling in this hook. Then it just uses an observer to fetch a value from the store and keep it updated.